### ISSUE TITLE
Panic when using <Input> inside <Suspense> under load

### ISSUE THREAD

**Issue by jerhat** [thaw-ui/thaw#349]
**Describe the bug:**
The application panics with the error message: `Dropped SendWrapper variable from a thread different to the one it has been created with.`

When accessing the below code I get a panic in the server. I managed to generate a minimal reproduction. Some changes stop reproducing the panic, like removing the `<Suspense>`, or using `<input>` instead of `<Input>` (see notes below).

Cargo dependencies and features:

```toml
[dependencies]
leptos = { version = "0.7.2" }
leptos_router = { version = "0.7.2" }
axum = { version = "0.7", optional = true }
console_error_panic_hook = "0.1"
leptos_axum = { version = "0.7.2", optional = true }
leptos_meta = { version = "0.7.2" }
tokio = { version = "1", features = ["rt-multi-thread"], optional = true }
tower = { version = "0.4", optional = true }
tower-http = { version = "0.5", features = ["fs"], optional = true }
wasm-bindgen = "0.2"
thiserror = "1"
http = "1"
thaw = { version = "0.4.0" }

[features]
hydrate = ["leptos/hydrate", "thaw/hydrate"]
ssr = [
    "dep:axum",
    "dep:tokio",
    "dep:tower",
    "dep:tower-http",
    "dep:leptos_axum",
    "leptos/ssr",
    "leptos_meta/ssr",
    "leptos_router/ssr",
    "thaw/ssr",
]
```

**Steps to reproduce the behavior:**

1. Use the start-axum-0.7 template
2. Switch the `app.rs` to use the following components:

```rust
use components::{Route, Router, Routes};
use leptos::prelude::Get;
use leptos::prelude::*;
use leptos_meta::{provide_meta_context, MetaTags};
use leptos_router::StaticSegment;
use leptos_router::*;
use thaw::*;

pub fn shell(options: LeptosOptions) -> impl IntoView {
    view! {
        <!DOCTYPE html>
        <html lang="en">
            <head>
                <meta charset="utf-8"/>
                <meta name="viewport" content="width=device-width, initial-scale=1"/>
                <AutoReload options=options.clone() />
                <HydrationScripts options/>
                <MetaTags/>
            </head>
            <body>
                <App/>
            </body>
        </html>
    }
}

#[server]
pub async fn submit_value(value: String) -> Result<(), ServerFnError> {
    use leptos::logging::log;
    log!("Received value: {:?}", value);
    Ok(())
}

#[server]
pub async fn get_counter() -> Result<u32, ServerFnError> {
    use tokio::time::{sleep, Duration};
    sleep(Duration::from_millis(100)).await;
    Ok(42)
}

#[component]
pub fn App() -> impl IntoView {
    provide_meta_context();

    view! {
        <ConfigProvider>
            <Router>
                <main>
                    <Routes fallback=|| "Page not found.".into_view()>
                        <Route path=StaticSegment("/") view=RootPage />
                        <Route path=StaticSegment("/form") view=|| view! { <UserForm counter=42 /> } />
                    </Routes>
                </main>
            </Router>
        </ConfigProvider>
    }
}

#[component]
fn RootPage() -> impl IntoView {
    let counter = Resource::new(|| (), move |_| get_counter());
    view! {
        <Suspense fallback=move || {
            view! { <div>"Loading"</div> }
        }>
        { move || Suspend::new(async move {
            let counter = counter.get().unwrap_or(Ok(0)).unwrap_or(0);
            view! {
                <UserForm counter=counter />
            }
        })}
        </Suspense>
    }
}

#[component]
fn UserForm(counter: u32) -> impl IntoView {
    let submit = ServerAction::<SubmitValue>::new();
    view! {
        <ActionForm action=submit>
            <Input name="value" value={"sometext".to_string()}></Input>
            // Replacing the above `<Input>` with the below `<input>` makes the pb go away
            //<input type="text" value="sometext" name="value" />

            <input type="submit" value="send" />
        </ActionForm>
        <div>{counter}</div>
    }
}
```

3. generate some load, for example using [hey](https://github.com/rakyll/hey):

```bash
$ hey http://127.0.0.1:3000
[...]
Status code distribution:
  [200] 170 responses

Error distribution:
  [30]  Get "http://127.0.0.1:3000": EOF
```

Here for example 30 requests out of 200 failed.

4. Observe the server logs:
`Dropped SendWrapper<T> variable from a thread different to the one it has been created with`

**Expected behavior**
All 200 requests get a 200 status code and no error in the server logs

**Additional notes**
- The error does not occur when using `<input />` instead of Thaw's `<Input />`.
- The error also does not occur when the form is not inside a `<Suspense />`.  i.e. the below command works as expected:

```bash
$ hey http://127.0.0.1:3000/form
[...]
Status code distribution:
  [200] 200 responses
```

- a similar issue exists in leptos repository: https://github.com/leptos-rs/leptos/issues/2907 . Since there is no panic when using `<input>`, I opened the issue here, but the root cause may actually be in leptos.

**Comment by luoxiaozero** [comment]
I fixed this issue in #352. Can you try it in your project?

```
thaw = { git = "https://github.com/thaw-ui/thaw", rev = "6313381" }
```

**Comment by jerhat** [comment]
I confirm this is working now, no more panics. Thank you.
